api/src/main/java/jakarta/data/repository/BasicRepository.java

```diff
@@ -148,6 +148,19 @@ public interface BasicRepository<T, K> extends DataRepository<T, K> {
     @Find
     Optional<T> findById(@By(ID) K id);
 
+    /**
+     * Retrieves the entities with the given unique identifiers.
+     * The returned list of entities is sorted by unique identifier
+     * in ascending order. If any of the unique identifiers is not
+     * found in the database, the returned list will be smaller
+     * than the supplied list of unique identifiers.
+     *
+     * @param ids a list of unique identifiers.
+     * @return the entities with the given unique identifiers.
+     */
+    @Query("where id(this) in :ids order by id(this)")
+    List<T> findByIdIn(@Param("ids") List<K> ids);
+
```

Comment:
The rest of this PR looks good, but I don't think we can add a new method to the existing BasicRepository interface now that we have released 1.0.  This would be a breaking change and would require at least an update of the major version.

```suggestion
```

Reply:
OK